pkg/kubecost/asset.go

```diff
@@ -2907,28 +2925,48 @@ func (asr *AssetSetRange) Append(that *AssetSet) {
 }
 
 // Each invokes the given function for each AssetSet in the range
+/*
 func (asr *AssetSetRange) Each(f func(int, *AssetSet)) {
 	if asr == nil {
 		return
 	}
 
+
 	for i, as := range asr.assets {
 		f(i, as)
 	}
 }
+*/
+
+// Each invokes the given function for each AssetSet in the range
+func (asr *AssetSetRange) Each(f func(int, *AssetSet)) {
+	if asr == nil {
+		return
+	}
+
+	asr.RLock()
+	var assets = make([]*AssetSet, len(asr.assets))
+	copy(assets, asr.assets)
+	asr.RUnlock()
```

Comment:
While I'm not crazy about duplicating the collection, it at least provides collection safety. Since deep copy is out of the question due to performance, this is the next best thing. Otherwise, we are opening the door for the collection to be modified during the each iteration (potentially by the callback func itself).